test/api.js

```diff
@@ -1,19 +1,200 @@
+                share_url: 'https://uni-muenster.sciebo.de/index.php/s/tY6I8NrDxTeXG85',
+                path: '/',
+                content_type: 'compendium_v1',
+            };
+
+            let j = request.jar();
+            let ck = request.cookie('connect.sid=' + cookie);
+            j.setCookie(ck, host);
+
+            request({
+                uri: host + '/api/v1/public-share',
+                method: 'POST',
+                jar: j,
+                form: form,
+                timeout: requestTimeout
+            }, (err, res, body) => {
+                assert.ifError(err);
+                assert.equal(res.statusCode, 403);
+                assert.isUndefined(JSON.parse(body).id, 'returned no id');
+                assert.propertyVal(JSON.parse(body), 'error', 'workspace creation not implemented');
+                done();
+            });
+        }).timeout(10000);
+
+        it('invalid share URL: should respond with an error 404', (done) => {
+            let form = {
+                share_url: 'https://uni-muenster.sciebo.de/index.php/s/7EoWgjLSFVV89AO not valid',
+                path: '/',
+                content_type: 'compendium_v1',
+            };
+
+            let j = request.jar();
+            let ck = request.cookie('connect.sid=' + cookie);
+            j.setCookie(ck, host);
+
+            request({
+                uri: host + '/api/v1/public-share',
+                method: 'POST',
+                jar: j,
+                form: form,
+                timeout: requestTimeout
+            }, (err, res, body) => {
+                assert.ifError(err);
+                assert.equal(res.statusCode, 404);
+                assert.isUndefined(JSON.parse(body).id, 'returned no id');
+                assert.propertyVal(JSON.parse(body), 'error', 'public share URL is invalid');
+                done();
+            });
+        }).timeout(10000);
+
+        it('invalid token: should respond with an error 404', (done) => {
+            let form = {
+                share_url: 'https://uni-muenster.sciebo.de/index.php/s/89k3ljf93kjfa',
+                path: '/',
+                content_type: 'compendium_v1',
+            };
+
+            let j = request.jar();
+            let ck = request.cookie('connect.sid=' + cookie);
+            j.setCookie(ck, host);
+
+            request({
+                uri: host + '/api/v1/public-share',
+                method: 'POST',
+                jar: j,
+                form: form,
+                timeout: requestTimeout
+            }, (err, res, body) => {
+                assert.ifError(err);
+                assert.equal(res.statusCode, 404);
+                assert.isUndefined(JSON.parse(body).id, 'returned no id');
+                assert.propertyVal(JSON.parse(body), 'error', 'could not read webdav contents');
+                done();
+            });
+        }).timeout(10000);
+
+        it('invalid webdav path: should respond with an error 404', (done) => {
```

Comment:
What about other ways the URL can be invalid? Please add checks for ...

- not a sciebo share
- not a valid URL